src/Strategy/Gradle/Errors.hs

```diff
@@ -0,0 +1,36 @@
+module Strategy.Gradle.Errors (
+  GradleCmdErrCtx (..),
+  FailedToListProjects (..),
+  refGradleDocUrl,
+) where
+
+import App.Docs (strategyLangDocUrl)
+import Data.Text (Text)
+import Diag.Diagnostic (ToDiagnostic (renderDiagnostic))
+import Effect.Logger (viaShow)
+import Path
+import Prettyprinter (Pretty (pretty), indent, vsep)
+
+refGradleDocUrl :: Text
+refGradleDocUrl = strategyLangDocUrl "gradle/gradle.md"
+
+newtype FailedToListProjects = FailedToListProjects (Path Abs Dir) deriving (Eq, Ord, Show)
+instance ToDiagnostic FailedToListProjects where
+  renderDiagnostic (FailedToListProjects dir) =
+    vsep
+      [ "Found a gradle build manifest in " <> viaShow dir <> " but, could not list projects."
```

Comment:
Originally, there was a note about this message not being helpful.  I think we should adjust this message (and all error-specific messages, as we find them) to instruct the user to do something useful:

- Ensure that they can run `gradle projects` (or `gradlew projects`, or `gradlew.bat projects`).  If they're not correct configured (e.g. missing credentials), then this will show them that.
- Ensure that the gradle version we ran is the gradle version they expect to use.  If not, then their gradle wrapper may be failing, and we are using our fallback to the version of gradle on the PATH.  We don't do a great job of reporting equivalent fallbacks for gradle.

Either of these would have been very helpful on a ticket I had recently, but there may be more.